src/Event/ApiGateway/WebsocketEvent.php

```diff
@@ -0,0 +1,141 @@
+<?php declare(strict_types=1);
+
+namespace Bref\Event\ApiGateway;
+
+use Bref\Event\InvalidLambdaEvent;
+use Bref\Event\LambdaEvent;
+
+/**
+ * Represents a Lambda event when Lambda is invoked by ApiGateway websocket route.
+ */
+final class WebsocketEvent implements LambdaEvent
+{
+    /** @var array */
+    private $event;
+
+    /** @var string */
+    private $routeKey;
+
+    /** @var int */
+    private $eventType;
+
+    /** @var mixed */
+    private $body;
+
+    /** @var string */
+    private $connectionId;
+
+    /** @var string */
+    private $domainName;
+
+    /** @var string */
+    private $apiId;
+
+    /** @var string */
+    private $stage;
+
+    /** Event types */
+    public const EVENT_TYPE_CONNECT = 0;
+    public const EVENT_TYPE_DISCONNECT = 1;
+    public const EVENT_TYPE_MESSAGE = 2;
+
+    /**
+     * @param mixed $event
+     * @throws InvalidLambdaEvent
+     */
+    public function __construct($event)
+    {
+        if (
+            ! is_array($event) ||
+            ! isset($event['requestContext']['routeKey']) ||
+            ! isset($event['requestContext']['eventType']) ||
+            ! isset($event['requestContext']['connectionId']) ||
+            ! isset($event['requestContext']['domainName']) ||
+            ! isset($event['requestContext']['apiId']) ||
+            ! isset($event['requestContext']['stage']) ||
+            ! in_array(
+                $event['requestContext']['eventType'],
+                [
+                    'CONNECT',
+                    'DISCONNECT',
+                    'MESSAGE',
+                ],
+                true
+            )
+        ) {
+            throw new InvalidLambdaEvent('Websocket', $event);
+        }
+
+        $this->domainName = $event['requestContext']['domainName'];
+        $this->connectionId = $event['requestContext']['connectionId'];
+        $this->routeKey = $event['requestContext']['routeKey'];
+        $this->apiId = $event['requestContext']['apiId'];
+        $this->stage = $event['requestContext']['stage'];
+        $this->event = $event;
+
+        switch ($event['requestContext']['eventType']) {
+            case 'CONNECT':
+                $this->eventType = self::EVENT_TYPE_CONNECT;
+                break;
+
+            case 'DISCONNECT':
+                $this->eventType = self::EVENT_TYPE_DISCONNECT;
+                break;
+
+            case 'MESSAGE':
+                $this->eventType = self::EVENT_TYPE_MESSAGE;
+                break;
```

Comment:
Why not have `self::EVENT_TYPE_CONNECT` constants (and the other 2) have the string values (`'CONNECT'`, etc.)? That way no need to map events manually to an integer value.

Reply:
@mnapoli The reasoning behind this is to force developers to use the constants rather than passing around strings. I work with a large code base on a daily basis and what we saw when having string values for constants was that developers choose to just pass the string value rather than passing the constant. This meant that once we had to re-iterate on our code base it was near impossible to locate peoples that referenced the constants & resulted it methods being near impossible to refactor. Simply because we had no way of knowing how things was passed around.

Having a integer as a value for the constants sort of forces developers to use constants. Obviously they could still pass around the numeric values - but they're less obvious for what they are and people then tend to use the constant instead.

This was at least my observation on this.

Reply:
Right, but then we would need a `default` case in that switch: but what would we put in there?

If we put an exception, then we are not future-proof (someone will need to patch that method the day a new event is added by AWS). If we silently ignore it, like right now, then we have a non-initialized property and the getter will throw an error (which is not future-proof as well).

I'd rather take the same approach as these event classes: they all expose a `toArray()` method, so that if users want to access data in the event that is not exposed by the class, then they can.

And while I agree with you about not using strings directly, I am also weary of adding a layer of abstraction on top of what AWS provides. Because that's one more thing to maintain, explain, etc. I would agree here it's not a huge deal, but small things like that add up.

Reply:
@mnapoli You're right. Now that i look at it again there is no need for adding this extra layer of abstraction on top of the event data. I suppose simple getters will do just fine here. I'll remove the constants and change it to just supply the event type forwarded.

Reply:
@mnapoli Fixed now. Now it returns the string value - so it's just a simple getter for accessing what ApiGateway forwards in the event 👍